crates/oxc_linter/src/rules/react/no_render_return_value.rs

```diff
@@ -0,0 +1,129 @@
+                    if matches!(
+                        parent_node.kind(),
+                        AstKind::VariableDeclarator(_)
+                            | AstKind::ObjectProperty(_)
+                            | AstKind::ReturnStatement(_)
+                            | AstKind::AssignmentExpression(_)
+                    ) {
+                        ctx.diagnostic(NoRenderReturnValueDiagnostic(call_expr.span))
+                    }
+
+                    let is_arrow_function =
+                        ctx.scopes().get_flags(parent_node.scope_id()).contains(ScopeFlags::Arrow);
+
+                    if is_arrow_function {
+                        ctx.nodes().ancestors(parent_node.id()).skip(1).into_iter().find(
+                            |node_id| {
+                                parent_node = ctx.nodes().get_node(*node_id);
+                                matches!(parent_node.kind(), AstKind::ArrowExpression(_))
+                                    .then(|| {
+                                        ctx.diagnostic(NoRenderReturnValueDiagnostic(
+                                            call_expr.span,
+                                        ));
+                                    })
+                                    .is_some()
+                            },
+                        );
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        ("ReactDOM.render(<div />, document.body);", None),
+        (
+            "
+        	        let node;
+        	        ReactDOM.render(<div ref={ref => node = ref}/>, document.body);
+        	      ",
+            None,
+        ),
+        ("ReactDOM.render(<div ref={ref => this.node = ref}/>, document.body);", None),
+        ("React.render(<div ref={ref => this.node = ref}/>, document.body);", None),
+        ("React.render(<div ref={ref => this.node = ref}/>, document.body);", None),
+        ("var foo = React.render(<div />, root);", None),
+        ("var foo = render(<div />, root)", None),
+        ("var foo = ReactDom.renderder(<div />, root)", None),
+    ];
+
+    let fail = vec![
+        ("var Hello = ReactDOM.render(<div />, document.body);", None),
+        (
+            "
+        	        var o = {
+        	          inst: ReactDOM.render(<div />, document.body)
+        	        };
+        	      ",
+            None,
+        ),
+        (
+            "
+        	        function render () {
+        	          return ReactDOM.render(<div />, document.body)
+        	        }
+        	      ",
+            None,
+        ),
+        ("var render = (a, b) => ReactDOM.render(a, b)", None),
+        ("this.o = ReactDOM.render(<div />, document.body);", None),
+        ("var v; v = ReactDOM.render(<div />, document.body);", None),
+        // ("var inst = React.render(<div />, document.body);", None),
```

Comment:
Need help with this?

Reply:
Yes, Thank you!

Different `react` versions have different render methods, and I don't currently know where to get the `react` version from. You can see https://github.com/jsx-eslint/eslint-plugin-react/blob/master/lib/rules/no-render-return-value.js#L39-L46

Reply:
Shall we just target >= 15.0.0 and leave a comment on why it's commented out?